### ISSUE TITLE
Use C# 8.0

### ISSUE THREAD

**Issue by JanLenoch** [Kentico/xperience-training-13#24]
### Who

- [ ] Anonymous visitor
- [ ] Patient
- [ ] Doctor
- [ ] Content editor
- [ ] Marketer
- [ ] Administrator
- [x] Developer

### Needs what

To use productivity features of C# 7.x and 8.0:

* non-nullable reference types
* async streams
* pattern matching
* succinct switch syntax
* using static
* null-coallescing assignment
* ref local variables
* discards
* expression-bodied members
* tuple return types and deconstruction
* local functions

### Why

The features help in code reliability, developer productivity, and performance.

* https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7
* https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7-1
* https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7-2
* https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7-3
* https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8

### Issue expands upon

This issue expands upon another issue(s): #`issue number`

### Blockers

This issue is blocked by the following issue(s): #`issue number`
